File: backend/src/entities/Person.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, DeleteDateColumn, ManyToOne, JoinColumn } from "typeorm";
import { Company } from "./Company";

@Entity()
export class Person {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column({ nullable: true })
  firstName?: string;

  @Column({ nullable: true })
  lastName?: string;

  @Column({ nullable: true })
  email?: string;

  @Column({ nullable: true })
  phone?: string;

  @Column({ nullable: true })
  jobTitle?: string;

  @Column({ nullable: true })
  city?: string;

  @Column({ nullable: true })
  avatarUrl?: string;

  @Column({ nullable: true })
  linkedinLink?: string;

  @Column({ nullable: true })
  xLink?: string;

  @Column({ nullable: true })
  companyId?: string;

  @ManyToOne(() => Company, (company) => company.people)
  @JoinColumn({ name: "companyId" })
  company?: Company;

  @CreateDateColumn()
  createdAt!: Date;

  @UpdateDateColumn()
  updatedAt!: Date;

  @DeleteDateColumn()
  deletedAt?: Date;
}

File: server/app/models/person.py
from typing import Optional
from beanie import Document, Indexed, Link
from pydantic import EmailStr
from datetime import datetime
from .company import Company

class Person(Document):
    first_name: str
    last_name: str
    email: Indexed(EmailStr, unique=True)
    phone: Optional[str] = None
    mobile: Optional[str] = None
    job_title: Optional[str] = None
    department: Optional[str] = None
    company: Optional[Link[Company]] = None
    linkedin: Optional[str] = None
    avatar_url: Optional[str] = None
    is_primary_contact: bool = False
    notes: Optional[str] = None
    created_by: Optional[str] = None # User UUID
    created_at: datetime = datetime.utcnow()
    updated_at: datetime = datetime.utcnow()

    class Settings:
        name = "people"
